### PR TITLE
[DO NOT MERGE WITHOUT FURTHER NOTICE] Added staging infrastructure for orchestrator

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1218

This PR should be merged after working hours and must be directly followed by JJ's manual adjustment of Palo Alto rules and whatever manual actions are needed. This needs to be arranged with JJ upfront.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: queue-staging.tf [hunk 1]
```diff
@@ -0,0 +1,104 @@
+module "queue-staging-namespace" {
```
Comment (nitinprabhu) [inline queue-staging.tf:1]: you can use the same namespace but create a different queue

File: queue-staging.tf [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+resource "azurerm_key_vault_secret" "envelopes_staging_queue_listen_conn_str" {
+  name      = "envelopes-staging-queue-listen-connection-string"
+  value     = "${module.envelopes-staging-queue.primary_listen_connection_string}"
+  vault_uri = "${data.azurerm_key_vault.key_vault.vault_uri}"
+}
+
+resource "azurerm_key_vault_secret" "envelopes_staging_queue_max_delivery_count" {
+  name      = "envelopes-staging-queue-max-delivery-count"
+  value     = "${var.envelope_staging_queue_max_delivery_count}"
```
Comment (nitinprabhu) [inline queue-staging.tf:56]: can use same env var as non staging queue

File: queue-staging.tf (hunk 1 above)
Comment (apereverzin) [reply queue-staging.tf:1]: should not we clearly split staging and non staging?

Comment (nitinprabhu) [reply queue-staging.tf:1]: If we have a good reason then yes otherwise it is not needed

File: queue-staging.tf [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  value = "${module.envelopes-staging-queue.primary_send_connection_string}"
+}
+
+output "processed_envelopes_staging_queue_primary_listen_connection_string" {
+  value = "${module.processed-envelopes-staging-queue.primary_listen_connection_string}"
+}
+
+output "processed_envelopes_staging_queue_primary_send_connection_string" {
+  value = "${module.processed-envelopes-staging-queue.primary_send_connection_string}"
+}
```
Comment (nitinprabhu) [inline queue-staging.tf:100]: why do we need this ?

File: queue-staging.tf [hunk 4]
```diff
@@ -0,0 +1,75 @@
+module "envelopes-staging-queue" {
+  source              = "git@github.com:hmcts/terraform-module-servicebus-queue?ref=master"
+  name                = "envelopes"
```
Comment (lgonczar) [inline queue-staging.tf:3]: Is this staging queue any different in its definition than the non-staging one?

File: queue-staging.tf [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+  requires_duplicate_detection            = "true"
+  duplicate_detection_history_time_window = "PT59M"
+  lock_duration                           = "PT5M"
+  max_delivery_count                      = "${var.envelope_queue_max_delivery_count}"
+}
+
+module "processed-envelopes-staging-queue" {
+  source              = "git@github.com:hmcts/terraform-module-servicebus-queue?ref=master"
+  name                = "processed-envelopes"
```
Comment (lgonczar) [inline queue-staging.tf:15]: Same as for the envelopes queue. Is it an exact duplicate of the non-staging queue?

File: queue-staging.tf [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  name      = "envelopes-staging-queue-listen-connection-string"
+  value     = "${module.envelopes-staging-queue.primary_listen_connection_string}"
+  vault_uri = "${data.azurerm_key_vault.key_vault.vault_uri}"
+}
+
+resource "azurerm_key_vault_secret" "envelopes_staging_queue_max_delivery_count" {
+  name      = "envelopes-staging-queue-max-delivery-count"
+  value     = "${var.envelope_queue_max_delivery_count}"
+  vault_uri = "${data.azurerm_key_vault.key_vault.vault_uri}"
+}
```
Comment (lgonczar) [inline queue-staging.tf:49]: I think we can use just one setting for this, no need for another secret. Staging and non-staging things should be configured identically.

File: queue-staging.tf [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  namespace_name      = "${module.queue-namespace.name}"
+  resource_group_name = "${azurerm_resource_group.rg.name}"
+  lock_duration       = "PT5M"
+  max_delivery_count  = "${var.payment_queue_max_delivery_count}"
+
+  duplicate_detection_history_time_window = "PT15M"
+}
+
+# region connection strings and other shared queue information as Key Vault secrets
+resource "azurerm_key_vault_secret" "envelopes_queue_staging_send_conn_str" {
```
Comment (lgonczar) [inline queue-staging.tf:33]: To be consistent with the rest of the names, it should be `envelopes_staging_queue_send_conn_str`

**Comment by apereverzin** [comment]
PR-169 includes these changes

**Pull request closed without merging**